Assets/IPieceState.cs

```diff
@@ -0,0 +1,18 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class IPieceState : MonoBehaviour
```

Comment:
Instead of class, let's change this to an interface with the interface keyword:
`public interface IPieceState`

MonoBehaviour is a specific Unity class that is used for elements that are accessible from Unity. Knowing when to extend from monobehaviour is an advanced kind of topic, but for the sake of making an interface, they typically don't extend from a parent class.

Here's an MSDN document on interfaces:
https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/types/interfaces